app/models/driver.rb

```diff
@@ -0,0 +1,35 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :nullify 
+  has_many :passengers, through: :trips
+
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  def average_rating
+    counter = 0
+    rating = 0
+    self.trips.each do |trip|
+      rating += trip.rating
+      counter += 1
+    end
+    return (rating.to_f/counter).round(2)
```

Comment:
This is pretty good -- a more programmer way of looking at this problem is:

1. Instead of `counter` and incrementing it, just using `self.trips.size` or `self.trips.count` to get you the size of the array
1. Instead of using a `.each`, using the Enumerables `sum` method https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-sum

This method could be as slim as:
```ruby
def average_rating
  return ( self.trips.sum / self.trips.size ).round(2)
end
```